examples/BASIC/BASIC.ino

```diff
@@ -514,16 +513,21 @@ static void updatePm(void) {
     Serial.printf("PM2.5 ug/m3: %d\r\n", measurements.pm25_1);
     Serial.printf("PM10 ug/m3: %d\r\n", measurements.pm10_1);
     Serial.printf("PM0.3 Count: %d\r\n", measurements.pm03PCount_1);
-    pmFailCount = 0;
+    ag.pms5003.resetFailCount();
   } else {
-    pmFailCount++;
-    Serial.printf("PMS read failed: %d\r\n", pmFailCount);
-    if (pmFailCount >= 3) {
+    ag.pms5003.updateFailCount();
+    Serial.printf("PMS read failed: %d\r\n", ag.pms5003.getFailCount());
+    if (ag.pms5003.getFailCount() >= PMS_FAIL_COUNT_SET_INVALID) {
       measurements.pm01_1 = utils::getInvalidPMS();
       measurements.pm25_1 = utils::getInvalidPMS();
       measurements.pm10_1 = utils::getInvalidPMS();
       measurements.pm03PCount_1 = utils::getInvalidPMS();
     }
+
+    if(ag.pms5003.getFailCount() >= ag.pms5003.getFailCountMax()) {
+      Serial.printf("Restarting...");
```

Comment:
Propose to write why we restart, e.g. something like
PMS failure count is %d, the max is set to %d, restarting ....